span.go

```diff
@@ -93,15 +81,20 @@ func (tx *Transaction) StartSpanOptions(name, spanType string, opts SpanOptions)
 	// Lock the parent first to avoid deadlocks in breakdown metrics calculation.
 	if opts.parent != nil {
 		opts.parent.mu.Lock()
-		defer opts.parent.mu.Unlock()
 	}
 
 	// Prevent tx from being ended while we're starting a span.
 	tx.mu.RLock()
 	defer tx.mu.RUnlock()
 	if tx.ended() {
+		if opts.parent != nil {
+			opts.parent.mu.Unlock()
+		}
```

Comment:
Moving the unlock away from the lock scares me a bit. TBH I'm really not a fan of this implicit exit span status, so I've raised the question with agent devs whether it is necessary to implement that part of the spec.

Reply:
:+1: makes sense. I'm not a fan of this locking either, so let's see what happens.